test/integration/sessions_test.rb

```diff
@@ -127,7 +127,7 @@ def setup
     user = FactoryBot.create(:user, account: @buyer, username: 'xi@example.net', password: 'wwwwww')
     user.activate
 
-    Authentication::Strategy::Internal.any_instance.expects(:authenticate_with_sso).with('yabadabado', '2016').returns(user)
+    Authentication::Strategy::Token.any_instance.expects(:authenticate_with_sso).with('yabadabado', '2016').returns(user)
```

Comment:
I assume this was a mistake in the original test. Perhaps change also test name to `with impersonate` instaed of `with sso`. Just a nit.

Reply:
It was not a mistake. Before there was a hierarchy of classes, and `Internal` was acting as a base class and implemented a few methods not related to internal auth at all. Methods were only there for the descendants.

About, `authenticate_with_sso`, I think I prefer not to do it because the module we use to check the token is called `ThreeScale::SSO::Encryptor`

https://github.com/jlledom/porta/blob/3fe9c2b368a129ca5685fad50b13825add6c38c8/app/lib/authentication/strategy/token.rb#L38

So `authenticate_with_sso` is not inaccurate IMO. What I think is we should rename `ThreeScale::SSO` to `ThreeScale::Token`. But that's probably out of scope for this PR.